.github/workflows/workflow.yml

```diff
@@ -55,9 +35,97 @@ jobs:
             exit 1
           fi
 
-      - name: Use OCaml ${{ matrix.ocaml-version }}
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        job:
+          - { os: macos-10.15, ocaml-version: 4.05.0 }
+          - { os: macos-10.15, ocaml-version: 4.08.1 }
+          - { os: macos-10.15, ocaml-version: "4.10.1+flambda" }
+          - { os: ubuntu-18.04, ocaml-version: 4.05.0 }
+          - { os: ubuntu-18.04, ocaml-version: "4.07.1+musl+flambda" }
+          - { os: ubuntu-18.04, ocaml-version: 4.08.1 }
+          - { os: ubuntu-18.04, ocaml-version: "4.08.1+flambda" }
+          - { os: ubuntu-18.04, ocaml-version: "4.08.1+fp+flambda" }
+          - { os: ubuntu-18.04, ocaml-version: "4.08.1+musl+flambda" }
+          - { os: ubuntu-18.04, ocaml-version: "4.10.1+musl+flambda" }
+          - { os: ubuntu-20.04, ocaml-version: 4.05.0 }
+          - { os: ubuntu-20.04, ocaml-version: "4.07.1+musl+flambda" }
+          - { os: ubuntu-20.04, ocaml-version: 4.08.1 }
+          - { os: ubuntu-20.04, ocaml-version: "4.08.1+musl+flambda" }
+          - { os: ubuntu-20.04, ocaml-version: "4.10.0+32bit" }
+          - { os: ubuntu-20.04, ocaml-version: "4.10.1+fp+flambda" }
+          - { os: ubuntu-20.04, ocaml-version: "4.10.1+musl+flambda" }
+          - { os: ubuntu-20.04, ocaml-version: "4.10.1+musl+static+flambda" }
+          - { os: windows-2019, ocaml-version: 4.05.0 }
+          - { os: windows-2019, ocaml-version: 4.08.1 }
+          - { os: windows-2019, ocaml-version: "4.08.1+mingw32c" }
+          - { os: windows-2019, ocaml-version: "4.10.1+mingw32c" }
+          - { os: windows-2019, ocaml-version: "4.10.1+mingw64c" }
+          - { os: windows-2019, ocaml-version: "4.10.1+msvc32c" }
+          - { os: windows-2019, ocaml-version: "4.10.1+msvc64c" }
+
+    runs-on: ${{ matrix.job.os }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Initialize workflow variables
+        id: vars
+        shell: bash
+        run: |
+          outputs() { for var in "$@" ; do echo steps.vars.outputs.${var}="${!var}" ; echo ::set-output name=${var}::${!var} ; done ; }
+          MINGW_ARCH='x86_64' ; case '${{ matrix.job.ocaml-version }}' in *+mingw32*) MINGW_ARCH='i686' ;; *+mingw64*) MINGW_ARCH='x86_64' ;; esac
+          MSVC_ARCH='' ; case '${{ matrix.job.ocaml-version }}' in *+msvc32*) MSVC_ARCH='x86' ;; *+msvc64*) MSVC_ARCH='x64' ;; esac
+          outputs MINGW_ARCH MSVC_ARCH
```

Comment:
```suggestion
          outputs() {
            for var in "$@"; do
              echo steps.vars.outputs."${var}"="${!var}"
              echo ::set-output name="${var}"::"${!var}"
            done
          }
          MINGW_ARCH='x86_64'
          case '${{ matrix.job.ocaml-version }}' in
            *+mingw32*) "$MINGW_ARCH"='i686' ;;
            *+mingw64*) "$MINGW_ARCH"='x86_64' ;;
          esac
          MSVC_ARCH=''
          case '${{ matrix.job.ocaml-version }}' in
            *+msvc32*) "$MSVC_ARCH"='x86' ;;
            *+msvc64*) "$MSVC_ARCH"='x64' ;;
          esac
          outputs "$MINGW_ARCH" "$MSVC_ARCH"
```

Reply:
Will do.

Reply:
The last line use of outputs is incorrect. `outputs MINGW_ARCH MSVC_ARCH` is the correct usage (variable names not values) and is left as is...